src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,3540 @@
+            LogInfo
+        );
+
+        if (LogInfo->stopRun) {
+            goto freeMem; // Exit function prematurely due to error
+        }
+
+    } else {
+        domCount[1] = fillCountX[0] = SW_Domain->nDimX;
+        domCount[0] = fillCountY[0] = SW_Domain->nDimY;
+        numVars = 4;
+
+        fillVals[2] = &valsYBnds;
+        fillVals[3] = &valsXBnds;
+        fillVarIDs[2] = YBndsID;
+        fillVarIDs[3] = XBndsID;
+    }
+
+    // --- Fill horizontal coordinate values
+
+    // Calculate resolution for y and x and allocate value arrays
+    resY = (SW_Domain->max_y - SW_Domain->min_y) / numY;
+    resX = (SW_Domain->max_x - SW_Domain->min_x) / numX;
+
+    // Generate/fill the horizontal coordinate variable values
+    // These values are within a bounding box given by the user
+    // I.e., lat values are within [ymin, ymax] and lon are within [xmin, xmax]
+    // bounds are ordered consistently with the associated coordinate
+    for (gridNum = 0; gridNum < numX; gridNum++) {
+        valsX[gridNum] = SW_Domain->min_x + (gridNum + .5) * resX;
+
+        if (!domTypeIsSite) {
+            bndsIndex = gridNum * 2;
+            valsXBnds[bndsIndex] = SW_Domain->min_x + gridNum * resX;
+            valsXBnds[bndsIndex + 1] = SW_Domain->min_x + (gridNum + 1) * resX;
+        }
+    }
+
+    for (gridNum = 0; gridNum < numY; gridNum++) {
+        valsY[gridNum] = SW_Domain->min_y + (gridNum + .5) * resY;
+
+        if (!domTypeIsSite) {
+            bndsIndex = gridNum * 2;
+            valsYBnds[bndsIndex] = SW_Domain->min_y + gridNum * resY;
+            valsYBnds[bndsIndex + 1] = SW_Domain->min_y + (gridNum + 1) * resY;
+        }
+    }
+
+    // --- Write out all values to file
+    for (varNum = 0; varNum < numVars; varNum++) {
+        SW_NC_write_vals(
+            &fillVarIDs[varNum],
+            domFileID,
+            NULL,
+            *fillVals[varNum],
+            start,
+            fillCounts[varNum],
+            "double",
+            LogInfo
+        );
+
+        if (LogInfo->stopRun) {
+            goto freeMem; // Exit function prematurely due to error
+        }
+    }
+
+    // Fill domain variable with SUIDs
+    SW_NC_write_vals(
+        &domID,
+        domFileID,
+        NULL,
+        domVals,
+        start,
+        domCount,
+        "unsigned integer",
+        LogInfo
+    );
+
+// Free allocated memory
+freeMem:
+    dealloc_netCDF_domain_vars(
+        &valsY, &valsX, &valsYBnds, &valsXBnds, &domVals
+    );
+}
+
+/**
+@brief Fill the variable "domain" with it's attributes
+
+@param[in] domainVarName User-specified domain variable name
+@param[in,out] domID Identifier of the "domain"
+    (or another user-specified name) variable
+@param[in] domDims Set dimensions for the domain variable
+@param[in] readinGeoYName User-provided geographical y-axis name
+@param[in] readinGeoXName User-provided geographical x-axis name
+@param[in] readinProgYName User-provided projected y-axis name
```

Comment:
Please correct `readinProgXName` and `readinProgYName` to `readinProjXName` and `readinProjYName` -- the "proj" refers to "projected"